plugins/router/template/router.go

```diff
@@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 	return router, nil
 }
 
+func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	if len(alias.PreferPort) == 0 {
+		return svc.EndpointTable
+	}
+	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
+	for i := range svc.EndpointTable {
+		endpoint := svc.EndpointTable[i]
+		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
```

Comment:
Right, it filters to the preferred port.  So if I define multiple ports in my route it doesn't do anything with the 0+n ports.  Just a bit confusing that I can configure for multiple ports and it ignores them.  It doesn't seem to be a reasonable use case to want 80 and 8888 of your pod on the same route anyway but if we did want to support it we'd need to plumb through the route.Ports.   Just another reason to get rid of this internal representation.